### PR TITLE
♻️ [Objectarium] Use binary hash as object identifiers

### PR DESCRIPTION
**Pull request by bdeneux**
Implement #184. See issue for more details.

### CODE REVIEW

File: contracts/okp4-objectarium/src/crypto.rs [hunk 1]
```diff
@@ -29,35 +35,145 @@ impl HashAlgorithm {
...
+    }
+}
+
+impl From<&Hash> for Vec<u8> {
+    fn from(hash: &Hash) -> Self {
+        hash.0.clone()
+    }
+}
+
+impl From<Hash> for Cow<'_, Hash> {
```
Comment (amimart) [inline contracts/okp4-objectarium/src/crypto.rs:111]: I think we can remove this implementation, in the other direction as well :)

**Review by amimart — CHANGES_REQUESTED** [review]
That's a nice piece thx! 💪 

I just pointed out a little something :)

**Review by amimart — APPROVED** [review]

**Review by ccamel — APPROVED** [review]
LGTM! Thanks 👍

**Comment by ccamel** [comment]
@bdeneux your branch is in conflict. Can you rebase? thanks 🙏

**Pull request merged**